### PR TITLE
Feature/cbft extended vote 642

### PR DESCRIPTION
**Pull request by scottmillner**

### CODE REVIEW

File: crates/consensus/authority/src/comet_bft/abci.rs [hunk 1]
```diff
@@ -312,4 +327,133 @@ where
...
+    #[error("invalid version")]
+    /// Invalid ExtendedVote version
+    InvalidVersion,
+}
+
+pub struct ExtendedVote {
+    pub version: Version,
+    pub bitcoin_block_hash: bitcoin::hash_types::BlockHash,
+    pub aggregated_public_key: secp256k1::PublicKey,
+}
```
Comment (arminsabouri) [inline crates/consensus/authority/src/comet_bft/abci.rs]: How do you feel about this living in its own file under the comet_bft dir

File: crates/consensus/authority/src/comet_bft/abci.rs [hunk 2]
```diff
@@ -312,4 +327,133 @@ where
...
+    #[error("invalid data format")]
+    /// Invalid data format
+    Decoding(#[from] encode::Error),
+    #[error("invalid version")]
+    /// Invalid ExtendedVote version
+    InvalidVersion,
+}
+
+pub struct ExtendedVote {
+    pub version: Version,
```
Comment (arminsabouri) [inline crates/consensus/authority/src/comet_bft/abci.rs]: can this just be u16?

Comment (scottmillner) [reply crates/consensus/authority/src/comet_bft/abci.rs]: yeah maybe a bit overkill with the struct but i like having a default method i can call so if the version is ever changed, i only have to update the default method...i use it in tests as well

made it u16 and added version_default() on ExtendedVote

File: crates/consensus/authority/src/comet_bft/abci.rs (hunk 1 above)
Comment (scottmillner) [reply crates/consensus/authority/src/comet_bft/abci.rs]: done!

**Review by arminsabouri — APPROVED** [review]

**Comment by arminsabouri** [comment]
@scottmillner I hope you don't mind I truncated some of these commits. Things like fixiing git conflicts

**Pull request merged**